### ISSUE TITLE
DAQ as a readable device

### ISSUE THREAD

**Issue by ZLLentz** [pcdshub/pcdsdaq#26]
## Expected Behavior

I can't remember why we opted against this, but I'm imagining something like:
```
from bluesky.plans import scan
from bluesky.run_engine import RunEngine

from pcdsdaq.daq import Daq
from pcdsdevices.device_types import IMS

RE = RunEngine({})
daq = Daq()
mot = IMS('XPP:USR:MMS:01', name='my_stage')
RE(scan([daq], mot, 0, 10, 11))
```
Where:
- `daq.stage` would start a new run
- `daq.trigger` would start acquiring for the configured time, and mark triggering as done when we've stopped
- `daq.read` would show information about how many events have passed (and maybe we'd find a good way to extract live feedback into here)
- `daq.read_configuration` would also show us the run number
- `daq.unstage` would end the run

I feel like this would be much simpler for the purposes of LCLS. We'd be able to use most of the built-in tools as-is.

As an aside, this makes it easier to implement scan PVs (it is revealed, the true reason I'm suggesting this...)

## Current Behavior

Only usable as a flyer, instead of as both a flyer and a reader

**Comment by teddyrendahl** [comment]
Can I be a 🍑 and say I suggested this back at the beginning? 😝 

I like this. The result is that you can just pass the `DAQ` in as a detector which seems the most intuitive thing for me. The only thing that is awkward is the `stage` `unstage` thing. If I want to have all my steps be different runs do I stage / unstage on every `per_step` hook? I guess that isn't crazy...

**Comment by ZLLentz** [comment]
Sorry for not remembering! Time has shown that using this a flyer is "correct" mechanically but unintuitive. IIRC `stage_wrapper` puts itself at every `open_run` and `close_run`, which I why I suggested that. So as long as you use the `stage_wrapper` and start/stop your runs at the right times, the `daq` will know how to partition the runs!

**Comment by ZLLentz** [comment]
Memory was bad, this is not how `stage_wrapper` works. Oh well.

**Comment by ZLLentz** [comment]
closed by #27
